internal/provider/validators/string_validators.go

```diff
@@ -102,3 +103,7 @@ func HecHostPortValidation() validator.String {
 func AuthenticationTokenValidation() validator.String {
 	return stringvalidator.RegexMatches(AuthenticationTokenRegex, "must be a valid authentication token")
 }
+
+func ApiKeyValidation() validator.String {
```

Comment:
```suggestion
func CrowdstrikeApiKeyValidation() validator.String {
```
Different API Keys are going to have different formats/requirements.
I'd recommend calling this specifically CrowdstrikeApiKeyValidation to distinguish it